File: TrainTimeWatch/Models/GPSQuality.swift
import SwiftUI

enum GPSQuality {
    case unavailable
    case lastKnown
    case poor
    case good

    var color: Color {
        switch self {
        case .unavailable: return .red
        case .lastKnown: return .gray
        case .poor: return .orange
        case .good: return .green
        }
    }

    static func from(accuracy: Double?) -> GPSQuality {
        guard let accuracy = accuracy else { return .unavailable }
        if accuracy < 0 { return .unavailable }
        if accuracy > 100 { return .poor }
        if accuracy > 30 { return .poor }
        return .good
    }
}

File: android/app/src/main/kotlin/com/evanjt/traintime/data/model/GpsQuality.kt
package com.evanjt.traintime.data.model

import androidx.compose.ui.graphics.Color

enum class GpsQuality {
    UNAVAILABLE,
    LAST_KNOWN,
    POOR,
    GOOD;

    val color: Color
        get() = when (this) {
            UNAVAILABLE -> Color.Red
            LAST_KNOWN -> Color.Gray
            POOR -> Color(0xFFFFA500)
            GOOD -> Color.Green
        }

    companion object {
        // Cached coordinates are reported as LAST_KNOWN by the caller, not here.
        fun from(accuracy: Double?): GpsQuality = when {
            accuracy == null -> UNAVAILABLE
            accuracy < 0 -> UNAVAILABLE
            accuracy > 30 -> POOR
            else -> GOOD
        }
    }
}
